pkg/pillar/zedUpload/commonutils/types.go

```diff
@@ -0,0 +1,33 @@
+// Copyright(c) 2017-2018 Zededa, Inc.
+// All rights reserved.
+
+package commonutils
+
+import (
+	"fmt"
+)
+
+// ChunkData contains the details of Chunks being downloaded
+type ChunkData struct {
+	Size  int64  // complete size to upload/download
+	Chunk []byte // chunk data being uploaded/downloaded
+}
+
+// ChunkDataChan is the uploading/downloading progress notification channel
+type ChunkDataChan chan ChunkData
+
+// ChunkWriter encapsulates the Chunk writing parameters
+type ChunkWriter struct {
+	Data          ChunkData
+	ChunkNotifier ChunkDataChan
+}
+
+// WriteAt is the overridden method of io.WriterAt method, which helps in chunk by chunk downloading.
+func (r *ChunkWriter) WriteAt(p []byte, off int64) (int, error) {
+	if r.ChunkNotifier != nil {
+		fmt.Println("s3 download", len(p))
```

Comment:
Where do you expect this println to be logged? In general we don't have packages write on stdout.

Reply:
I had added this for debugging purpose, I have cleaned them up.